rero_ils/modules/holdings/cli.py

```diff
@@ -0,0 +1,106 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Click command-line interface for item record management."""
+
+from __future__ import absolute_import, print_function
+
+import json
+
+import click
+from flask.cli import with_appcontext
+
+from ..documents.api import DocumentsSearch
+from ..holdings.api import create_holding
+from ..item_types.api import ItemTypesSearch
+from ..locations.api import LocationsSearch
+from ..organisations.api import Organisation
+
+
+def get_documents_with_type_journal():
+    """Get pids of documents with type journal."""
+    document_pids = []
+    es_documents = DocumentsSearch()\
+        .filter('term', type="journal").source(['pid']).scan()
+    for es_document in es_documents:
+        document_pids.append(es_document.pid)
+    return document_pids
+
+
+def get_location(library_pid):
+    """Rertun a location for the given library pid."""
+    results = LocationsSearch().source(['pid'])\
+        .filter('term', library__pid=library_pid)\
+        .scan()
+    locations = [location.pid for location in results]
+    return locations[0]
+
+
+def get_circ_category(org_pid):
+    """Get random circ category for an organisation pid."""
+    results = ItemTypesSearch().source(['pid'])\
+        .filter('term', organisation__pid=org_pid)\
+        .scan()
+    records = [record.pid for record in results]
+    return records[0]
+
+
+def get_random_location(org_pid):
+    """Return random location for an organisation pid."""
+    org = Organisation.get_record_by_pid(org_pid)
+    libraries = [library.pid for library in org.get_libraries()]
+    return get_location(libraries[0])
+
+
+@click.command('create_patterns')
+@click.option('-v', '--verbose', 'verbose', is_flag=True, default=False)
+@click.option('-d', '--debug', 'debug', is_flag=True, default=False)
+@click.argument('infile', type=click.File('r'))
+@with_appcontext
+def create_patterns(infile, verbose, debug):
+    """Create serials patterns for documents of type journals.
+
+    infile: Json patterns file
+    """
+    click.secho('Create serials patterns:', fg='green')
+    journal_pids = get_documents_with_type_journal()
+    data = json.load(infile)
+    record_index = 0
+    for record in data:
+        template_name = record.get('template_name')
+        try:
+            document_pid = journal_pids[record_index]
+        except IndexError as error:
+            break
+        patterns = record.get('patterns')
+        for org_pid in Organisation.get_all_pids():
+            circ_category_pid = get_circ_category(org_pid)
+            location_pid = get_random_location(org_pid)
+            holding_pid = create_holding(
```

Comment:
If we change return value of `create_holding` we have to write:

holding_pid = create_holding(...).pid